tests/test_common_gpu.py

```diff
@@ -1083,6 +1085,85 @@ def test_8bit_dora_merging(self):
         assert torch.allclose(out_dora, out_unmerged, atol=atol, rtol=rtol)
         assert torch.allclose(out_dora, out_unloaded, atol=atol, rtol=rtol)
 
+    @require_torch_gpu
+    @pytest.mark.single_gpu_tests
+    def test_dora_ephemeral_transfers(self):
+        torch.manual_seed(0)
+        model = AutoModelForCausalLM.from_pretrained(
+            "facebook/opt-350m",
```

Comment:
I ran the tests on a 4090 and unfortunately, it is very flaky for me, failing in 4 out of 5 tries with

> assert 2.1404892549999204 > (1.1 * 1.980610250000609)

and similar numbers. I also tried a bigger model, `bloomz-560m`, but that still failed. I wonder if we should scrap the performance test if the gain doesn't materialize on smaller models (and we don't want to load huge models during testing either).

Not sure what we can really test instead, maybe just that there is no error and the device, similar to the multi-GPU test?

As a "proof" that this works, I would suggest to instead add a short script to the `examples/` directory that uses a bigger model where there is a very noticeable gain. WDYT?

Reply:
Yes, that seems to be a better approach. My intention is to expand this in other areas post-merge. Hopefully one of those is more obvious.

I'll rewrite this to check that the resulting weights are identical instead, which ensures that this code does not screw anything up.

Reply:
Added example that loads a DoRA onto a model, and optionally merges and saves it to disk, while also timing it.

The one annoying part is that it will obviously not get the time right if the model is not yet in the HF hub cache and has to be downloaded first. I might look into whether I can force a "download only if necessary" call before the actual model/DoRA load part.

Reply:
I put in `snapshot_download` calls before the timers. It downloads some unnecessary stuff but those aren't very large, and it looks like it works otherwise.